tests/helpers.rs

```diff
@@ -0,0 +1,111 @@
+//! Helpers for tests which exercise the overall application, in particular
+//! the `MDBook` initialization and build/rendering process.
+//!
+//! This will create an entire book in a temporary directory using some
+//! dummy content.
+
+#![allow(dead_code, unused_variables, unused_imports)]
+extern crate tempdir;
+
+use std::path::Path;
+use std::fs::{self, File};
+use std::io::{Read, Write};
+
+use tempdir::TempDir;
+
+
+const SUMMARY_MD: &'static str = "# Summary
+
+[Introduction](intro.md)
+
+- [First Chapter](./first/index.md)
+    - [Nested Chapter](./first/nested.md)
+- [Second Chapter](./second.md)
+
+[Conclusion](./conclusion.md)
+";
+
+const INTRO: &'static str = "# Introduction
+
+Here's some interesting text...";
+
+const FIRST: &'static str = "# First Chapter
+
+more text.";
+
+const NESTED: &'static str = r#"# Nested Chapter
+
+This file has some testable code.
+
+```rust
+assert!($TEST_STATUS);
+```"#;
+
+const SECOND: &'static str = "# Second Chapter";
+
+const CONCLUSION: &'static str = "# Conclusion";
+
+
+/// Create a dummy book in a temporary directory, using the contents of
+/// `SUMMARY_MD` as a guide.
+///
+/// The "Nested Chapter" file contains a code block with a single
+/// `assert!($TEST_STATUS)`. If you want to check MDBook's testing
+/// functionality, `$TEST_STATUS` can be substitute for either `true` or
+/// `false`. This is done using the `passing_test` parameter.
+pub fn create_book(passing_test: bool) -> TempDir {
```

Comment:
I like the `BookBuilder` idea! It'll make things a lot easier in the long run, and also integrates quite nicely with your earlier dummy book directory comment.